sys/net/routing/ng_rpl/ng_rpl_control_messages.c

```diff
@@ -0,0 +1,603 @@
+            if (ng_rpl_dodags[i].state != 0) {
+                ng_rpl_dodags[i].dodag_conf_counter = 0;
+                ng_rpl_send_DIO(&ng_rpl_dodags[i], src);
+            }
+        }
+    }
+}
+
+bool _parse_options(ng_rpl_dodag_t *dodag, ng_rpl_opt_t *opt, uint16_t len, ng_ipv6_addr_t *src)
+{
+    uint16_t l = 0;
+    ng_rpl_opt_target_t *first_target = NULL;
+    while(l < len) {
+        if ((opt->type != NG_RPL_OPT_PAD1) && (len < opt->length + sizeof(ng_rpl_opt_t) + l)) {
+            /* return false to delete the dodag,
+             * because former options may also contain errors */
+            return false;
+        }
+
+        switch(opt->type) {
+            case (NG_RPL_OPT_PAD1): {
+                DEBUG("RPL: PAD1 option parsed\n");
+                l += 1;
+                opt = (ng_rpl_opt_t *) (((uint8_t *) opt) + 1);
+                continue;
+            }
+            case (NG_RPL_OPT_PADN): {
+                DEBUG("RPL: PADN option parsed\n");
+                break;
+            }
+            case (NG_RPL_OPT_DODAG_CONF): {
+                DEBUG("RPL: DODAG CONF DIO option parsed\n");
+                ng_rpl_opt_dodag_conf_t *dc = (ng_rpl_opt_dodag_conf_t *) opt;
+                ng_rpl_of_t *of = ng_rpl_get_of_for_ocp(byteorder_ntohs(dc->ocp));
+                if (of != NULL) {
+                    dodag->instance->of = of;
+                }
+                else {
+                    DEBUG("RPL: Unsupported OCP 0x%02x\n", byteorder_ntohs(dc->ocp));
+                    dodag->instance->of = ng_rpl_get_of_for_ocp(NG_RPL_DEFAULT_OCP);
+                }
+                dodag->dio_interval_doubl = dc->dio_int_doubl;
+                dodag->dio_min = dc->dio_int_min;
+                dodag->dio_redun = dc->dio_redun;
+                dodag->instance->max_rank_inc = byteorder_ntohs(dc->max_rank_inc);
+                dodag->instance->min_hop_rank_inc = byteorder_ntohs(dc->min_hop_rank_inc);
+                dodag->default_lifetime = dc->default_lifetime;
+                dodag->lifetime_unit = byteorder_ntohs(dc->lifetime_unit);
+                dodag->trickle.Imin = (1 << dodag->dio_min);
+                dodag->trickle.Imax = dodag->dio_interval_doubl;
+                dodag->trickle.k = dodag->dio_redun;
+                break;
+            }
+            case (NG_RPL_OPT_TARGET): {
+                DEBUG("RPL: RPL TARGET DAO option parsed\n");
+                kernel_pid_t if_id = ng_ipv6_netif_find_by_prefix(NULL, &dodag->dodag_id);
+                if (if_id == KERNEL_PID_UNDEF) {
+                    DEBUG("RPL: no interface found for the configured DODAG id\n");
+                    return false;
+                }
+
+                ng_rpl_opt_target_t *target = (ng_rpl_opt_target_t *) opt;
+                if (first_target == NULL) {
+                    first_target = target;
+                }
+
+                fib_add_entry(if_id, target->target.u8, sizeof(ng_ipv6_addr_t), AF_INET6, src->u8,
```

Comment:
Maybe this can be left as a follow up PR. I believe there needs to be some extra handling inside this parser for adding routing entries for entire networks (non-128 prefix length).